test_conformance/basic/test_vector_swizzle.cpp

```diff
@@ -22,6 +22,8 @@
 #include "procs.h"
 #include "harness/testHarness.h"
 
+std::string pragma_extension;
```

Comment:
Wouldn't it be better to make this a local variable inside `test_vectype`?  You should then be able to use an `if constexpr` to set the extension pragma based on the template argument type.  That way we don't need any unnecessary global state.

Reply:
To be honest it crossed my mind but I deduced it is not worth the effort. The thing is that `constexpr` condition will confuse `cl_ushort` and `cl_half`, I would have to pass an additional template argument to inform `test_vectype` we are dealing with FP type which eventually makes the code less readable.